docs/modules/ROOT/pages/usage.adoc

```diff
@@ -0,0 +1,390 @@
+The first thing to notice is that the contract imports the interfaces of the implementations that will be customized.
+These will be used in the next code example.
+
+Next, the contract includes the {erc20-component} implementations; however, `ERC20Impl` and `ERC20CamelOnlyImplt` are *not* embedded.
+Creating a custom implementation of an interface means that the component implementation cannot be embedded.
+The following example shows the pausable logic integrated into the ERC20 implementations:
+
+[,javascript]
+----
+#[starknet::contract]
+mod ERC20Pausable {
+    (...)
+
+    // Custom ERC20 implementation
+    #[external(v0)]
+    impl CustomERC20Impl of IERC20<ContractState> {
+        fn transfer(
+            ref self: ContractState, recipient: ContractAddress, amount: u256
+        ) -> bool {
+            // Add the custom logic
+            self.pausable.assert_not_paused();
+            // Add the original implementation method from `IERC20Impl`
+            self.erc20.transfer(recipient, amount)
+        }
+
+        fn total_supply(self: @ContractState) -> u256 {
+            // This method's behavior does not change from the component
+            // implementation, but this method must still be defined.
+            // Simply add the original implementation method from `IERC20Impl`
+            self.erc20.total_supply()
+        }
+
+        (...)
+    }
+
+    // Custom ERC20CamelOnly implementation
+    #[external(v0)]
+    impl CustomERC20CamelOnlyImpl of IERC20CamelOnly<ContractState> {
+        fn totalSupply(self: @ContractState) -> u256 {
+            self.erc20.total_supply()
+        }
+
+        fn balanceOf(self: @ContractState, account: ContractAddress) -> u256 {
+            self.erc20.balance_of(account)
+        }
+
+        fn transferFrom(
+            ref self: ContractState,
+            sender: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) -> bool {
+            self.pausable.assert_not_paused();
+            self.erc20.transfer_from(sender, recipient, amount)
+        }
+    }
+}
+----
+
+Notice that in the `CustomERC20Impl`, the `transfer` method integrates `pausable.assert_not_paused` as well as `erc20.transfer` from `PausableImpl` and `ERC20Impl` respectively.
+This is why the contract defined the `ERC20Impl` from the component in the previous example.
+
+Creating a custom implementation of an interface must define *all* methods from that interface.
+This is true even if the behavior of a method does not change from the component implementation (as `total_supply` exemplifies in this example).
+
+TIP: The ERC20 documentation provides a more specific custom implementations guide with {custom-decimals}.
+
+=== Accessing component storage
+
+Just as contracts can access methods within a component implementation, contracts can also access component storage.
+Storage members are accessible to the contract by instantiating the component's `InternalImpl` like this:
```

Comment:
are you sure it's the internalimpl the one that enables this?

Reply:
Whoops 😅 should be substorage attribute^